src/shared/components/code-editor/__tests__/YAMLCodeEditor.spec.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { render, screen } from '@testing-library/react';
+import { useSwaggerDefinitions } from '../hooks/useSwaggerDefinitions';
+import { registerYAMLinMonaco } from '../monaco';
+import { YAMLCodeEditor } from '../YAMLCodeEditor';
+
+jest.mock('@monaco-editor/react', () => ({
+  loader: {
+    config: jest.fn(),
+    init: jest.fn().mockResolvedValue({}),
+  },
+  default: (props) => <div data-test="monaco-editor">{props.value}</div>,
+}));
+
+jest.mock('../hooks/useSwaggerDefinitions', () => ({
+  useSwaggerDefinitions: jest.fn(),
+}));
+
+jest.mock('../monaco', () => ({
+  registerYAMLinMonaco: jest.fn(),
+}));
+
+jest.mock('@patternfly/react-code-editor', () => ({
+  CodeEditor: (props) => {
+    React.useEffect(() => {
+      props.onEditorDidMount?.({}, {});
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+    return <pre data-test="mock-editor">{props.code}</pre>;
+  },
+  Language: { yaml: 'yaml', json: 'json' },
+  EditorDidMount: jest.fn(),
+}));
+
+const queryClient = new QueryClient();
+
+describe('YAMLCodeEditor', () => {
+  beforeEach(() => {
+    (useSwaggerDefinitions as jest.Mock).mockReturnValue({
+      data: {
```

Comment:
it seems the data is also enjoyed by ReleaseYamlTab.spec.tsx.  How about we create data file to ensure they are able to share the vars ?

Reply:
nice suggestion, updated!